src/main/java/com/mixpanel/android/mpmetrics/MixpanelPushNotification.java

```diff
@@ -277,31 +295,51 @@ protected void maybeAddActionButtons() {
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT_WATCH) {
             for (int i = 0; i < data.buttons.size(); i++) {
                 NotificationButtonData btn = data.buttons.get(i);
-                builder.addAction(this.createAction(btn.icon, btn.label, btn.uri));
+                builder.addAction(this.createAction(btn.icon, btn.label, btn.uri, btn.actionId));
             }
         }
     }
 
     @TargetApi(20)
-    protected Notification.Action createAction(int icon, CharSequence title, String uri) {
-        return (new Notification.Action.Builder(icon, title, createActionIntent(uri))).build();
+    protected Notification.Action createAction(int icon, CharSequence title, String uri, String actionId) {
+        return (new Notification.Action.Builder(icon, title, createActionIntent(uri, actionId))).build();
     }
 
-    protected PendingIntent createActionIntent(String uri) {
-        return PendingIntent.getActivity(
-                context,
-                0,
-                new Intent(Intent.ACTION_VIEW, Uri.parse(uri)),
-                PendingIntent.FLAG_UPDATE_CURRENT
-        );
+    protected PendingIntent createActionIntent(String uri, String actionId) {
```

Comment:
Perhaps add some javadoc to this method describing what it does, particularly for things like `FLAG_ACTIVITY_NO_HISTORY`.